spec/registry.md

```diff
@@ -66,8 +66,33 @@ the corresponding input and options rules.
 If multiple `<override>` elements would match the current locale,
 only the first one is used.
 
-Matching-function signatures additionally include one or more `<match>` elements
-to define the keys against which they can match when used as selectors.
+### Variant Key Matches
+
+Matching-function signatures can include `<matches>` and `<when>` elements
+defining the variant keys matched by the selector.
+
+Each `<matches>` MAY contain either one or more `<match>` elements, or an `href` attribute.
+If an `href` attribute is set, its URL value MUST resolve to an XML document
+with a root `<matches>` element with no `href` attribute,
+which will then replace the current `<matches>` element for all later processing.
+If `<matches>` contains any child elements, its `href` attribute is ignored.
+Otherwise, if `<matches>` contains an `href` attribute, its `validationRule` attribute is ignored.
+
+The `<match>` element whose `locales` best matches the current locale
+using resource item [lookup](https://unicode.org/reports/tr35/#Lookup) from LDML is used.
+An element with no `locales` attribute is the default
+(and is considered equivalent to the `root` locale).
+
+As the available keys may depend on option values,
+`<when>` elements can be used to select an appropriate `<matches>` element for selection.
+If the resolved or default value of a selector option
+corresponding to the `<when>` `option` attribute
+is included in its list of `values`,
+its contents are considered before any and all later `<when>` and `<matches>` elements.
+If a `<matches>` element within a `<when>` has a `<match>` for the current locale,
+later `<matches>` elements outside that `<when>` are not considered.
```

Comment:
This is really hard to grok.

I think maybe this should go sooner, since `<when>` (when present) is a parent to `<matches>`. Something like:

> The results of a selector depend on the the value of options.
> If a `<when>` element is present that matches a selector `option`
> then only the elements contained by the `<when>` are used to evaluate
> the selector.

Question: what about nesting of `when`?

Reply:
The nesting is handled perhaps a bit too implicitly by the `(when*,matches?)` ordering, and the use of the term "contents" referring to both in the passage above.

The language you propose is a bit problematic, because
1. It makes it possible for two different `<when>` elements to both qualify for being "used to evaluate
the selector."
2. It presumes completeness of inner `<matches>`. For example, with the plural integer selector that I've used as an example, the `<matches>` for `maximumFractionDigits=0` might not include anything for `en` because English doesn't make a distinction on that axis. Then we need to look at the plural selection `<matches>`, which would include an `en` entry.

But your overall concern about the language here being hard to grok is on point, and it should be revised.

Reply:
> If a `<matches>` element within a `<when>` has a `<match>` for the current locale, later `<matches>` elements outside that `<when>` are not considered.

Is this correct? Maybe the missing piece is:

```suggestion
If a `<matches>` element within a matching `<when>` has a `<match>` for the current locale,
later `<matches>` outside that `<when>` are not considered.
```

That is, we only want to consider this if the `when` elements options match, not for irrelevant `when` elements?

(Otherwise a good improvement)

Reply:
Yeah, that does improve things. Edited your suggestion to not drop the preceding 5 lines and just rephrase the last sentence.